File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {boolean}
     */
    constructor(){
        this.arr = []
    }

    isValidBST(root) {
        this.rec(root)
        console.log(this.arr)
        if(this.arr.length == 1){
            return true
        }

        for(let i = 0; i < this.arr.length - 1; i++){
            if(this.arr[i] < this.arr[i + 1]) continue;
            return false
        }   
        return true
    }     

    rec(root){
        let resL
        let resR
        if(root.left) resL = this.rec(root.left)
        this.arr.push(root.val)
        if(root.right) resR = this.rec(root.right)
    }
}

File: Data Structures & Algorithms/balanced-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        self.answer = True
        def recursion(node):
            if node == None:
                return 0
            print(node.val)
            left = recursion(node.left) + 1
            right = recursion(node.right) + 1
            if left - right > 1 or right - left > 1:
                self.answer = False
            return max(left, right)
        recursion(root)
        return self.answer

        